<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from standalone_mico.xacro          | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ADA" xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz" xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint" xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model" xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics" xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable" xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro" xmlns:xi="http://www.w3.org/2001/XInclude">
  <!-- Base link with continuous joint at the output -->
  <!-- Suggested parameters: mico link_base link_1 joint_1 -->
  <!-- Shoulder link with revolute joint at the output -->
  <!-- Suggested parameters: mico link_1 link_2 joint_2 -->
  <!-- Upper arm link with revolute joint at the output -->
  <!-- Suggested parameters: mico link_2 link_3 joint_3 -->
  <!-- Lower arm link with revolute joint at the output -->
  <!-- Suggested parameters: mico link_3 link_4 joint_4 -->
  <!-- Wrist link with continuous joint at the output -->
  <!-- Suggested parameters (for first link): mico link_4 link_5 joint_5 -->
  <!-- Wrist link with continuous joint at the output -->
  <!-- Suggested parameters (for first link): mico link_4 link_5 joint_5 -->
  <!-- Joint-Link pair for a single finger. -->
  <!-- Suggested parameters: jaco2 jaco2_link_hand 1 (origin xyz) (origin rpy) -->
  <!-- Hand link with three finger joints -->
  <!-- Suggested parameters (for first link): mico link_hand -->
  <!-- Base link model -->
  <link name="mico_link_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.07625"/>
      <mass value="1.000"/>
      <inertia ixx="2.5e-3" ixy="0.00" ixz="0.00" iyy="2.50e-3" iyz="0.00" izz="1.25e-3"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_base.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_base.dae"/>
      </geometry>
    </collision>
  </link>
  <!-- First rotation (yaw) -->
  <joint name="mico_joint_1" type="continuous">
    <parent link="mico_link_base"/>
    <child link="mico_link_1"/>
    <axis xyz="0 0 -1"/>
    <limit effort="30" velocity="0.84"/>
    <origin rpy="3.14159265359 0 3.14159265359" xyz="0 0 0.1535"/>
  </joint>
  <transmission name="mico_joint_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_1">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_1_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- Base link model -->
  <link name="mico_link_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0077 0.000 -0.0728"/>
      <mass value="0.825"/>
      <inertia ixx="6.18e-4" ixy="4.54e-8" ixz="9.82e-4" iyy="5.72e-4" iyz="-1.26e-8" izz="2.58e-4"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_1.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_1.dae"/>
      </geometry>
    </collision>
  </link>
  <!-- Second rotation (shoulder pitch) -->
  <joint name="mico_joint_2" type="revolute">
    <parent link="mico_link_1"/>
    <child link="mico_link_2"/>
    <origin rpy="-1.57079632679 3.14159265359 0" xyz="0 0 -0.1185"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-3.8451785" upper="0.7035858" velocity="0.8425"/>
  </joint>
  <transmission name="mico_joint_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_2">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_2_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- Base link model -->
  <link name="mico_link_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.1441 0.000 -0.0261"/>
      <mass value="1.215"/>
      <inertia ixx="4.33e-4" ixy="2.31e-7" ixz="-1.61e-6" iyy="7.18e-3" iyz="8.58e-8" izz="7.44e-3"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_2.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_2.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="mico_joint_3" type="revolute">
    <parent link="mico_link_2"/>
    <child link="mico_link_3"/>
    <origin rpy="3.14159265359 0 1.57079632679" xyz="0.29 0 0"/>
    <axis xyz="0 0 -1"/>
    <limit effort="30" lower="-4.104090" upper="0.95768524" velocity="0.83"/>
  </joint>
  <transmission name="mico_joint_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_3">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_3_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- Base link model -->
  <link name="mico_link_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0396 0.0000 -0.0215"/>
      <mass value="0.775"/>
      <inertia ixx="2.28e-4" ixy="-5.09e-8" ixz="-3.87e-5" iyy="3.88e-4" iyz="6.63e-8" izz="4.42e-4"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_3.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_3.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="mico_joint_4" type="continuous">
    <parent link="mico_link_3"/>
    <child link="mico_link_4"/>
    <origin rpy="3.14159265359 1.57079632679 0" xyz="0.123 0 -0.00845"/>
    <axis xyz="0 0 -1"/>
    <limit effort="30" velocity="0.84"/>
  </joint>
  <transmission name="mico_joint_4_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_4">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_4_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- Base link model -->
  <link name="mico_link_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0083 0.0000 -0.0378"/>
      <mass value="0.415"/>
      <inertia ixx="5.45e-5" ixy="3.27e-8" ixz="-1.77e-8" iyy="5.53e-5" iyz="-5.56e-6" izz="4.87e-5"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_4.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_4.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="mico_joint_5" type="continuous">
    <parent link="mico_link_4"/>
    <child link="mico_link_5"/>
    <origin rpy="0 -1.0471975512 0" xyz="0.037 0 -0.06408"/>
    <axis xyz="0 0 -1"/>
    <limit effort="30" velocity="0.84"/>
  </joint>
  <transmission name="mico_joint_5_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_5">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_5_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- Base link model -->
  <link name="mico_link_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0083 0.0000 -0.0378"/>
      <mass value="0.415"/>
      <inertia ixx="5.45e-5" ixy="3.27e-8" ixz="-1.77e-8" iyy="5.53e-5" iyz="-5.56e-6" izz="4.87e-5"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_4.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_4.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="mico_joint_6" type="continuous">
    <parent link="mico_link_5"/>
    <child link="mico_link_hand"/>
    <origin rpy="-1.0471975512 0 1.57079632679" xyz="0.037 0 -0.06408"/>
    <axis xyz="0 0 -1"/>
    <limit effort="30" velocity="0.84"/>
  </joint>
  <transmission name="mico_joint_6_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_6">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_6_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- Base link model -->
  <link name="mico_link_hand">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0034 0.0000 -0.0808"/>
      <mass value="0.557"/>
      <inertia ixx="1.51e-6" ixy="-1.90e-10" ixz="-4.87e-9" iyy="1.14e-5" iyz="1.52e-8" izz="1.05e-5"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_hand.dae"/>
        <!-- <mesh filename="package://ada_description/meshes/Hand_Link_camera.STL"/>  -->
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/mico_link_hand.dae"/>
        <!-- <mesh filename="package://ada_description/meshes/Hand_Link_camera.STL"/>  -->
      </geometry>
    </collision>
  </link>
  <link
    name="Camera_Depth_Frame">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="0.05" />
      <inertia
        ixx="0"
        ixy="0"
        ixz="0"
        iyy="0"
        iyz="0"
        izz="0" />
    </inertial>
  </link>
   <joint
    name="Camera_Joint" type = "fixed">
    <origin
      xyz="-0.047 0.05 -0.090 "
      rpy="-2.92 0.000 0" />
    <parent
      link="mico_link_hand" />
    <child
      link="Camera_Depth_Frame" />
  </joint>
  <joint
    name="Camera_Depth_Joint" type = "fixed">
    <origin
      xyz="0.045 -0.015 -0.03"
      rpy="0 0.000 0" />
    <parent
      link="Camera_Depth_Frame" />
    <child
      link="Camera_RGB_Frame" />
  </joint>
   <link
    name="Camera_RGB_Frame">
        <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="0.05" />
      <inertia
        ixx="0"
        ixy="0"
        ixz="0"
        iyy="0"
        iyz="0"
        izz="0" />
    </inertial>
  </link>
  
  <link
    name="Camera_Collision_Frame">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="0.05" />
      <inertia
        ixx="0"
        ixy="0"
        ixz="0"
        iyy="0"
        iyz="0"
        izz="0" />
    </inertial>
    <visual>
      <geometry>
        <!--<mesh filename="package://ada_description/meshes/StructureIO.stl"/>-->
        <mesh filename="package://ada_description/meshes/structureio.dae"/>
        <!-- <mesh filename="package://ada_description/meshes/Hand_Link_camera.STL"/>  -->
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/StructureIO.stl"/>
        <!-- <mesh filename="package://ada_description/meshes/Hand_Link_camera.STL"/>  -->
      </geometry>
    </collision>
  </link>
   <joint
    name="Camera_Collision_Joint" type = "fixed">
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <parent
      link="mico_link_hand" />
    <child
      link="Camera_Collision_Frame" />
  </joint>
  <link name="mico_link_finger_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.020 -0.015 0"/>
      <mass value="0.010"/>
      <inertia ixx="5.00e-5" ixy="0" ixz="0" iyy="1.58e-5" iyz="0" izz="1.58e-4"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_proximal_limb_1.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_proximal_limb_1.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="mico_joint_finger_1" type="revolute">
    <parent link="mico_link_hand"/>
    <child link="mico_link_finger_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="1.57079632679 1.57079632679 0" xyz="-0.03095 -0.0025 -0.116"/>
    <limit effort="30" lower="0" upper="1.35" velocity="0.785398163"/>
  </joint>
  <transmission name="mico_joint_finger_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_finger_1">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_finger_1_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="mico_link_finger_tip_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.020 -0.015 0"/>
      <mass value="0.010"/>
      <inertia ixx="5.00e-7" ixy="0" ixz="0" iyy="1.58e-6" iyz="0" izz="1.58e-6"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_distal_limb_1.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_distal_limb_1.dae"/>
      </geometry>
    </collision>
  </link>
  <!-- The tip joint is fixed for now, but the revolute version is
             available under it -->
  <joint name="mico_joint_finger_tip_1" type="fixed">
    <parent link="mico_link_finger_1"/>
    <child link="mico_link_finger_tip_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.035 -0.0295 -0.0025"/>
  </joint>
  <!--
        <joint name="${prefix}_joint_finger_tip_${suffix}" type="revolute">
            <parent link="${prefix}_link_finger_${suffix}"/>
            <child link="${prefix}_link_finger_tip_${suffix}"/>
            <axis xyz="0 0 1"/>
            <origin xyz="0.035 -0.0295 -0.0025" rpy="0 0 0"/>
            <limit lower="0" upper="${J_PI/180*40}" effort="30" velocity="5"/>
         </joint>
         <transmission name="${prefix}_joint_finger_tip_${suffix}_trans">
            <type>transmission_interface/SimpleTransmission</type>
            <joint name="${prefix}_joint_finger_tip_${suffix}">
                <hardwareInterface>EffortJointInterface</hardwareInterface>
            </joint>
            <actuator name="${prefix}_joint_finger_tip_${suffix}_motor">
                <mechanicalReduction>1.0</mechanicalReduction>
            </actuator>
        </transmission>
        -->
  <link name="mico_link_finger_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.020 -0.015 0"/>
      <mass value="0.010"/>
      <inertia ixx="5.00e-5" ixy="0" ixz="0" iyy="1.58e-5" iyz="0" izz="1.58e-4"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_proximal_limb_1.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_proximal_limb_1.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="mico_joint_finger_2" type="revolute">
    <parent link="mico_link_hand"/>
    <child link="mico_link_finger_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.57079632679 1.57079632679 0" xyz="0.03095 -0.0025 -0.116"/>
    <limit effort="30" lower="0" upper="1.35" velocity="0.785398163"/>
  </joint>
  <transmission name="mico_joint_finger_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="mico_joint_finger_2">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="mico_joint_finger_2_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="mico_link_finger_tip_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.020 -0.015 0"/>
      <mass value="0.010"/>
      <inertia ixx="5.00e-7" ixy="0" ixz="0" iyy="1.58e-6" iyz="0" izz="1.58e-6"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_distal_limb_1.dae"/>
      </geometry>
      <material name="carbon_fiber">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ada_description/meshes/finger_distal_limb_1.dae"/>
      </geometry>
    </collision>
  </link>
  <!-- The tip joint is fixed for now, but the revolute version is
             available under it -->
  <joint name="mico_joint_finger_tip_2" type="fixed">
    <parent link="mico_link_finger_2"/>
    <child link="mico_link_finger_tip_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.035 -0.0295 -0.0025"/>
  </joint>
  <!--
        <joint name="${prefix}_joint_finger_tip_${suffix}" type="revolute">
            <parent link="${prefix}_link_finger_${suffix}"/>
            <child link="${prefix}_link_finger_tip_${suffix}"/>
            <axis xyz="0 0 1"/>
            <origin xyz="0.035 -0.0295 -0.0025" rpy="0 0 0"/>
            <limit lower="0" upper="${J_PI/180*40}" effort="30" velocity="5"/>
         </joint>
         <transmission name="${prefix}_joint_finger_tip_${suffix}_trans">
            <type>transmission_interface/SimpleTransmission</type>
            <joint name="${prefix}_joint_finger_tip_${suffix}">
                <hardwareInterface>EffortJointInterface</hardwareInterface>
            </joint>
            <actuator name="${prefix}_joint_finger_tip_${suffix}_motor">
                <mechanicalReduction>1.0</mechanicalReduction>
            </actuator>
        </transmission>
        -->
  <!-- End effector offset -->
  <joint name="mico_link_hand_to_end_effector" type="fixed">
    <parent link="mico_link_hand"/>
    <child link="mico_end_effector"/>
    <origin rpy="0 3.14159265359 0" xyz="0 0 -0.1687"/>
  </joint>
  <link name="mico_end_effector"/>
  <!-- Used to fix the model to the ground in Gazebo. -->
  <link name="world"/>
  <joint name="world_2_base_fixed" type="fixed">
    <parent link="world"/>
    <child link="mico_link_base"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <gazebo reference="mico_link_base">
    <material>Gazebo/Black</material>
    <selfCollide>false</selfCollide>
    <gravity>false</gravity>
  </gazebo>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control"/>
  </gazebo>
</robot>
